src/test/java/org/mockitousage/debugging/VerificationListenerCallBackTest.java

```diff
@@ -0,0 +1,123 @@
+package org.mockitousage.debugging;
+
+import org.junit.Test;
+import org.mockito.exceptions.base.MockitoAssertionError;
+import org.mockito.exceptions.misusing.NullInsteadOfMockException;
+import org.mockito.internal.progress.MockingProgress;
+import org.mockito.internal.progress.ThreadSafeMockingProgress;
+import org.mockito.internal.verification.api.VerificationData;
+import org.mockito.listeners.VerificationListener;
+import org.mockito.verification.VerificationMode;
+import org.mockito.verification.VerificationSucceededEvent;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.*;
+
+public class VerificationListenerCallBackTest {
+    public static final String invocationWanted = "doSomething";
+
+    @Test
+    public void should_call_single_listener_on_verify() {
+        //given
+        RememberingListener listener = new RememberingListener();
+        ThreadSafeMockingProgress.mockingProgress().addListener(listener);
+        Foo foo = mock(Foo.class);
+
+        //when
+        VerificationMode never = never();
+        verify(foo, never).doSomething("");
+
+        //then
+        assertThatHasBeenNotified(listener, foo, never, invocationWanted);
+    }
+
+    @Test
+    public void should_call_all_listeners_on_verify() {
+        //given
+        RememberingListener listener1 = new RememberingListener();
+        RememberingListener listener2 = new RememberingListener();
+
+        MockingProgress mockingProgress = ThreadSafeMockingProgress.mockingProgress();
+        mockingProgress.addListener(listener1);
+        mockingProgress.addListener(listener2);
+
+        Foo foo = mock(Foo.class);
+
+        //when
+        VerificationMode never = never();
+        verify(foo, never).doSomething("");
+
+        //then
+        assertThatHasBeenNotified(listener1, foo, never, invocationWanted);
+        assertThatHasBeenNotified(listener2, foo, never, invocationWanted);
+    }
+
+    @Test
+    public void should_not_call_listener_when_verify_was_called_incorectly() {
+        //given
+        RememberingListener listener = new RememberingListener();
+        MockingProgress mockingProgress = ThreadSafeMockingProgress.mockingProgress();
+
+        mockingProgress.addListener(listener);
+
+        //when
+        Foo foo = null;
+        try {
+            verify(foo).doSomething("");
+            fail("Exception expected.");
+        } catch (Exception e) {
+            //then
+            assertNull(listener.cause);
```

Comment:
This way of verifying that a listener was not called seems fragile. Could we improve this?